code/_onclick/hud/action.dm

```diff
@@ -31,14 +32,16 @@
 		Remove(owner)
 	owner = T
 	T.actions += src
+	if (button)
+		button.moved = 0
 	T.update_action_buttons()
 
 /datum/action/proc/Remove(mob/living/T)
 	if(button)
 		if(T.client)
 			T.client.screen -= button
-		qdel(button)
-		button = null
+		T.contents -= button
```

Comment:
Mostly worried about things like xenos and space worms (do we still have those?)... Please test that contents doesn't make its way into the world.